test/utils.js

```diff
@@ -600,6 +601,72 @@ const formatExchangeObj = (srcAddr, destAddr, amount, wrapper, destAmount = 0, u
     ];
 };
 
+const formatExchangeObjCurve = async (
+    srcAddr,
+    destAddr,
+    amount,
+    wrapper,
+) => {
+    const { route: sdkRoute } = await curve.getBestRouteAndOutput(
+        srcAddr,
+        destAddr,
+        '1000', // this is fine
+    );
+    const swapParams = sdkRoute.map((e) => [e.i, e.j, e.swapType]).concat(
+        [...Array(4 - sdkRoute.length).keys()].map(
+            () => [0, 0, 0],
+        ),
+    );
+    const route = [srcAddr].concat(
+        ...sdkRoute.map((e) => [e.poolAddress, e.outputCoinAddress]),
+        ...[...Array(8 - (sdkRoute.length) * 2).keys()].map(
+            () => [nullAddress],
+        ),
+    );
+
+    const exchangeData = hre.ethers.utils.defaultAbiCoder.encode(
+        ['address[9]', 'uint256[3][4]'],
+        [route, swapParams],
+    );
+
+    return [
+        srcAddr,
+        destAddr,
+        amount,
+        0,
+        0,
+        0,
+        nullAddress,
+        wrapper,
+        exchangeData,
+        [nullAddress, nullAddress, nullAddress, 0, 0, hre.ethers.utils.toUtf8Bytes('')],
+    ];
+};
+
+const formatExchangeObj = async (srcAddr, destAddr, amount, wrapper, destAmount = 0, uniV3fee) => {
```

Comment:
This will break other 50+ instances where we call formatExchangeObj without await, why not just pass additional param?